### PR TITLE
More helpful error message for pgf backend

### PR DESCRIPTION
**Pull request by nwin**
Makes the error message more helpful if subprocess fails with a file-not-found-error.

### CODE REVIEW

File: lib/matplotlib/backends/backend_pgf.py [hunk 1]
```diff
@@ -316,8 +317,14 @@ def __init__(self):
                                      stdin=subprocess.PIPE,
                                      stdout=subprocess.PIPE,
                                      cwd=self.tmpdir)
-        except OSError:
-            raise RuntimeError("Error starting process '%s'" % self.texcommand)
+        except OSError as e:
+            if e.errno == errno.ENOENT:
+                raise OSError(errno.ENOENT, "Latex command not found. "
```
Comment (tacaswell) [inline lib/matplotlib/backends/backend_pgf.py]: Is there a good reason to not re-raise this as an `RuntimeError`?  By changing the exception type we are breaking API (as there is probably user code which expects to catch `RuntimeError` here).

**Comment by tacaswell** [comment]
@pwuertz

File: lib/matplotlib/backends/backend_pgf.py (hunk 1 above)
Comment (pwuertz) [inline lib/matplotlib/backends/backend_pgf.py]: I guess the idea was to discriminate between the "command not found" and "command returned with error" case, so errors when running the latex process are propagated as RuntimeError, not finding the latex command is an OS or environment error. That distinction is a new "feature" if you would like to say so. I don't really know if this is to be considered as breaking the API, since the behaviour on error is neither defined nor documented anywhere.

**Comment by pwuertz** [comment]
I'm confused.. didn't I apply such a patch to master already? Can't find it though..

**Comment by pwuertz** [comment]
No wait, I was the one catching and re-raising the OSError as RuntimeError with an error string (#3158). Do you feel that these additional changes to the error string are necessary?

**Comment by nwin** [comment]
TLDR: I do not care which error type you want to have but I think it would be nice if we could provide a help if the cause of the error is almost certain.

Until “recently” (well F21 still ships an older version) it was an uncomprehensible uncatched error message and without stackoverflow I would not have any idea what as going on… http://stackoverflow.com/questions/20000427/matplotlib-pgf-oserror-no-such-file-or-directory-in-subprocess-py Because of that I also heavily doubt that anybody relies on the fact that the error is an RuntimeError.

Thus I thought it would be nice to discriminate between some random error an a specific error whose cause is known.

Actually I  think that this is a issue in python itself. It is not helpful at all to raise an file-not-found error without mentioning the requested file command.

**Comment by tacaswell** [comment]
Fair enough, target 1.5.x (no backporting).

I assumed that the `RuntimeError` was intentional because it was converting an `OSError`.

Can you add a note in https://github.com/matplotlib/matplotlib/tree/master/doc/api/api_changes  I agree it seems pedantic, but I think it is worth doing.

**Comment by nwin** [comment]
I thought about it, converting to RuntimeError is probably a good idea because people will maybe catch an OSError to check if their (data) files are missing. Sorry for the discussion I'll change it. ;)

**Comment by pwuertz** [comment]
Thanks for this enhancement!
